bazel/BUILD

```diff
@@ -35,6 +35,35 @@ genrule(
     stamp = 1,
 )
 
+config_setting(
+    name = "windows_x86_64",
+    values = {"cpu": "x64_windows"},
+)
+
+config_setting(
+    name = "windows_opt_build",
+    values = {
+        "cpu": "x64_windows",
+        "compilation_mode": "opt",
+    },
+)
+
+config_setting(
+    name = "windows_dbg_build",
+    values = {
+        "cpu": "x64_windows",
+        "compilation_mode": "dbg",
+    },
+)
+
+config_setting(
+    name = "windows_fastbuild_build",
```

Comment:
I'm not that familiar with Bazel's platforms/toolchains support, but is this something we should be using to make the Windows vs. Linux vs. Mac OS X situation cleaner? In particular, I think we have multiple orthogonal build concerns:
1. Target OS (Windows (native vs. Linux compat?), Linux, Mac OS X)
2. Target runtime (libc++ vs. libstdc++)
3. Target toolchain (clang, gcc, Visual Studio)
4. Etc.

Reply:
We added the `windows_*_build` config settings because there is no way to perform a `select` on multiple conditions. This allows `envoy_copts` to conditionally append `--ggdb3` on Linux and never to append it on Windows: https://github.com/greenhouse-org/envoy/blob/windows-build-pr/bazel/envoy_build_system.bzl#L36-L41
 
With respect to the more general question, we are open to suggestions and what the community decides as the best approach.

Reply:
@laszlocsomor @jmillikin-stripe is there a sensible way to handle this with https://docs.bazel.build/versions/master/platforms.html?

Reply:
I think adding a cflag on one platform should be a property of the cc_toolchain, if I understand it properly (given how confusing C++ is, I possibly don't). Are you defining your own cc_toolchain targets, or using the ones from auto-configuration?

Reply:
If I understand correctly, on Windows we are just using the ones from auto-configuration, though on Linux there's a wrapper script: https://github.com/envoyproxy/envoy/blob/master/bazel/cc_configure.bzl

Reply:
In that case, the existing code is probably the best way to do things.